packages/next/src/server/instrumentation/utils.ts

```diff
@@ -0,0 +1,12 @@
+export function getRevalidateReason(params: {
```

Comment:
I think we should align this with existing handling:
https://github.com/vercel/next.js/blob/8eb64ad515011575f3f0c32a454699cd29024700/packages/next/src/server/render.tsx#L851-L855

And test:
https://github.com/vercel/next.js/blob/canary/test/e2e/revalidate-reason/revalidate-reason.test.ts

Reply:
It's modified from here but stripped "build", since onRequestError doesn't cover errors from the build phase. They're considered as build erros.

So that in app-render I'm using `isRevalidate` state from static generation store to determine if it's doing ISR. We can't directly using that condition in app-render as we need to separate normal rendering from ISR rendering.

Or do you mean we just sort of unify the 2 implmentation into one place?

Reply:
Ah ok - I was initially thinking maybe the types and function for determining the different revalidation reasons could be shared, but maybe not necessary since instrumentation doesn't cover the build case.